pkg/git/libgit2/managed/options.go

```diff
@@ -18,35 +18,38 @@ package managed
 
 import (
 	"sync"
+
+	"github.com/fluxcd/source-controller/pkg/git"
 )
 
 // TransportOptions represents options to be applied at transport-level
 // at request time.
 type TransportOptions struct {
 	TargetURL string
-	CABundle  []byte
+	AuthOpts  *git.AuthOptions
 }
 
 var (
+	// transportOpts maps a unique id to a set of transport options.
 	transportOpts = make(map[string]TransportOptions, 0)
 	m             sync.RWMutex
 )
 
-func AddTransportOptions(targetUrl string, opts TransportOptions) {
+func AddTransportOptions(id string, opts TransportOptions) {
```

Comment:
We should document that the `id` must be a valid URL based on the protocols registered by the Managed Transport, otherwise this mechanism won't work.